lib/misc/firestore.dart

```diff
@@ -17,15 +53,56 @@ class Firestore {
       'streak': 0,
       'registered_on': Timestamp.now(),
       'total_distance': 0,
-      'challenges_completed': 0
+      'challenges_completed': 0,
+      'last_challenge_completed': null,
+      'challenge_pending': null,
     });
+  }
 
-    await FirebaseFirestore.instance
+  Future<bool> isChallengePending() async {
+    if (user == null) {
+      return false;
+    }
+    final docSnapshot = await FirebaseFirestore.instance
+        .collection('users')
+        .doc(user!.uid)
+        .get();
+    final challengePending = docSnapshot.data()?['challenge_pending'];
+
+    return challengePending != null;
```

Comment:
může se stát že challenge pending bude `false`? Pak by to vrátilo true, přestože hodnota je false